hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java

```diff
@@ -104,15 +103,73 @@ public void migrate(@NonNull final MigrationContext ctx) {
                             endpointFor(nodeInfo.externalHostName(), nodeInfo.externalPort())))
                     .gossipCaCertificate(nodeInfo.sigCertBytes())
                     .weight(nodeInfo.stake())
-                    .adminKey(finalAdminKey)
-                    .build();
+                    .adminKey(finalAdminKey);
+            if (nodeDetailMap != null) {
+                nodeDetail = nodeDetailMap.get(nodeInfo.nodeId());
+                if (nodeDetail != null) {
+                    nodeBuilder
+                            .serviceEndpoint(nodeDetail.serviceEndpoint())
+                            .grpcCertificateHash(nodeDetail.nodeCertHash());
+                }
+            }
             writableNodes.put(
-                    EntityNumber.newBuilder().number(nodeInfo.nodeId()).build(), node);
-        });
+                    EntityNumber.newBuilder().number(nodeInfo.nodeId()).build(), nodeBuilder.build());
+        }
 
         log.info("Migrated {} nodes from address book", addressBook.size());
     }
 
+    private Key getAccountAdminKey(@NonNull final MigrationContext ctx) {
+        var adminKey = Key.DEFAULT;
+
+        final var accountConfig = ctx.configuration().getConfigData(AccountsConfig.class);
+        ReadableKVState<AccountID, Account> readableAccounts = null;
+
+        try {
+            readableAccounts = ctx.newStates().get(ACCOUNTS_KEY);
+        } catch (IllegalArgumentException e) {
+            log.info("AccountStore is not found, can be ignored.");
+        }
```

Comment:
Why would AccountStore be not found ?

Reply:
It should be found. Just in case if not found(one of my test cases), we do use the follow key as adminKey.
 Key finalAdminKey = adminKey == null || adminKey.equals(Key.DEFAULT)
                ? Key.newBuilder().ed25519(bootstrapConfig.genesisPublicKey()).build()
                : adminKey;